backend/server/rhnVirtualization.py

```diff
@@ -532,14 +532,27 @@ def __db_insert_domain(self, host_id, uuid, properties):
             raise VirtualizationEventError('unable to get virt instance id')
         id = row['id']
 
+        # Do we have a system with a machine id matching the uuid?
+        get_system_id_sql = """
+            SELECT s.id
+            FROM rhnServer s
+            LEFT JOIN rhnVirtualInstance vi on vi.virtual_system_id = s.id
+            WHERE s.machine_id = ':uuid' and vi.virtual_system_id IS NULL
+        """
+        query = rhnSQL.prepare(get_system_id_sql)
+        query.execute(uuid=uuid)
+        row = query.fetchone_dict() or {}
+
+        guest_id = row['id'] if row and 'id' in row else 'null'
```

Comment:
Not sure if it is right to set it to a string 'null'.
You should try this one:

```suggestion
        guest_id = row['id'] if row and 'id' in row else None
```

AFAIK 'None' is converted to a database NULL.

Reply:
indeed, fixed it. Nice catch